docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc

```diff
@@ -1214,14 +1214,40 @@ If you work with a Quarkus OIDC `web-app` application, then the Quarkus OIDC cod
 To use the refresh token, you should carefully configure the session cookie age.
 The session age should be longer than the ID token lifespan and close to or equal to the refresh token lifespan.
 
-You calculate the session age by adding the lifespan value of the current ID token and the values of the `quarkus.oidc.authentication.session-age-extension` and `quarkus.oidc.token.lifespan-grace` properties.
+You calculate the session cookie age by adding the lifespan value of the current ID token and the values of the `quarkus.oidc.authentication.session-age-extension` and `quarkus.oidc.token.lifespan-grace` properties.
 
 [TIP]
 ====
-You use only the `quarkus.oidc.authentication.session-age-extension` property to significantly extend the session lifespan, if required.
 You use the `quarkus.oidc.token.lifespan-grace` property only to consider some small clock skews.
 ====
 
+[TIP]
+====
+You use the `quarkus.oidc.authentication.session-age-extension` property to significantly extend the session lifespan in order to
+make even expired ID token available to the application.
+
+Session cookie `max-age` property is set to the value of the ID token life-span by default and the authenticated user is redirected
+to the OIDC provider to re-authenticate, after the session cookie has expired and the user has attempted to access the application again.
+
+However, if `quarkus.oidc.authentication.session-age-extension` is set to a reasonable idle-time duration of several hours,
+longer than an ID token life-span, then, even if the ID token has expired, it can still be made available to the application which may refresh it
+or redirect the user to a session expired page, because the session cookie which keeps ID token is still available during the current authenticated user's request.
+
+For example, let's assume you expect a typical user activity to last for 30 minutes.
+Therefore, you have configured OIDC provider to issue ID tokens which are valid for 30 minutes only.
+But, you would also like to have a user's ID token refreshed and the session renewed after this user has returned from a one hour break.
+In this case, set this property to 90 minutes, otherwise, Quarkus will not be able to detect a session cookie, and the user will be required to re-authenticate.
+
+An important point is that extending the session cookie's age with `quarkus.oidc.authentication.session-age-extension`
+is only about maximizing the chances of making even an expired ID token associated with this cookie available to the application for further processing.
+
+It improves a user experience, otherwise, an authenticated user may be surprised by seeing an immediate re-authentication challenge
+if the ID token has expired together with the session cookie which holds it.
+
+By default, this property will be set to 1 hour if either `quarkus.oidc.token.refresh-expired` or `quarkus.oidc.authentication.session-expired-page`
```

Comment:
```suggestion
By default, this property will be set to 1 hour if either `quarkus.oidc.token.refresh-expired` or `quarkus.oidc.authentication.session-expired-page` features are enabled since they can only be effective if an expired ID token is available.
```